src/sinks/aws_s_s/sns/config.rs

```diff
@@ -0,0 +1,114 @@
+use aws_sdk_sns::Client as SnsClient;
+
+use crate::config::{
+    AcknowledgementsConfig, DataType, GenerateConfig, Input, ProxyConfig, SinkConfig, SinkContext,
+};
+use futures::FutureExt;
+use vector_config::configurable_component;
+
+use super::{client::SnsMessagePublisher, BaseSSSinkConfig, ConfigWithIds, SSSink};
+use crate::aws::create_client;
+use crate::aws::ClientBuilder;
+
+/// Configuration for the `aws_sns` sink.
+#[configurable_component(sink(
+    "aws_sns",
+    "Publish observability events to AWS Simple Queue Service topics."
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct SnsSinkConfig {
+    /// The ARN of the Amazon SNS topic to which messages are sent.
+    #[configurable(validation(format = "uri"))]
+    #[configurable(metadata(
+        docs::examples = "arn:aws:sns:us-east-2:123456789012:MyTopic"
+    ))]
+    pub topic_arn: String,
+
+    #[serde(flatten)]
+    pub base_config: BaseSSSinkConfig,
+}
+
+impl GenerateConfig for SnsSinkConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"topic_arn = "arn:aws:sns:us-east-2:123456789012:MyTopic"
+            region = "us-east-2"
+            encoding.codec = "json""#,
+        )
+        .unwrap()
+    }
+}
+
+impl SnsSinkConfig {
+    pub async fn healthcheck(self, client: SnsClient) -> crate::Result<()> {
+        client
+            .get_topic_attributes()
+            .topic_arn(self.topic_arn.clone())
+            .send()
+            .await
+            .map(|_| ())
+            .map_err(Into::into)
+    }
+
+    pub async fn create_client(&self, proxy: &ProxyConfig) -> crate::Result<SnsClient> {
+        create_client::<SnsClientBuilder>(
+            &self.base_config.auth,
+            self.base_config.region.region(),
+            self.base_config.region.endpoint(),
+            proxy,
+            &self.base_config.tls,
+            true,
+        )
+        .await
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "aws_sns")]
+impl SinkConfig for SnsSinkConfig {
+    async fn build(
+        &self,
+        cx: SinkContext,
+    ) -> crate::Result<(crate::sinks::VectorSink, crate::sinks::Healthcheck)> {
+        let client = self.create_client(&cx.proxy).await?;
+
+        let publisher = SnsMessagePublisher::new(client.clone(), self.topic_arn.clone());
+
+        let healthcheck = self.clone().healthcheck(client.clone()).boxed();
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I know this was from the previous code but I think it's unecessary for the healthcheck to be a part of the struct and require cloning like this. IMO its much cleaner to extract the healthcheck function and pass in the needed info which in this case is just the topic_arn and the client. The Clickhouse sink has an example of this

```Rust
let healthcheck = Box::pin(healthcheck(client.clone(), self.topic_arn.clone());
```

Reply:
done